Source/Core/Core/ConfigLoaders/GameConfigLoader.cpp

```diff
@@ -0,0 +1,415 @@
+                         [&](const auto& entry) { return entry.second == location; });
+
+  if (it != ini_to_location.end())
+    return it->first;
+
+  // Try again, but this time with an empty key
+  // Certain sections like 'Speedhacks' have keys that are variable
+  it = std::find_if(ini_to_location.begin(), ini_to_location.end(), [&](const auto& entry) {
+    return std::tie(entry.second.system, entry.second.key) ==
+           std::tie(location.system, location.key);
+  });
+  if (it != ini_to_location.end())
+    return it->first;
+
+  WARN_LOG(COMMON, "Unknown option: %s.%s", location.section.c_str(), location.key.c_str());
+  return {"", ""};
+}
+
+// INI Game layer configuration loader
+class INIGameConfigLayerLoader : public Config::ConfigLayerLoader
+{
+public:
+  INIGameConfigLayerLoader(const std::string& id, u16 revision, bool global)
+      : ConfigLayerLoader(global ? Config::LayerType::GlobalGame : Config::LayerType::LocalGame),
+        m_id(id), m_revision(revision)
+  {
+  }
+
+  void Load(Config::Layer* config_layer) override
+  {
+    IniFile ini;
+    if (config_layer->GetLayer() == Config::LayerType::GlobalGame)
+    {
+      for (const std::string& filename : GetGameIniFilenames(m_id, m_revision))
+        ini.Load(File::GetSysDirectory() + GAMESETTINGS_DIR DIR_SEP + filename, true);
+    }
+    else
+    {
+      for (const std::string& filename : GetGameIniFilenames(m_id, m_revision))
+        ini.Load(File::GetUserPath(D_GAMESETTINGS_IDX) + filename, true);
+    }
+
+    const std::list<IniFile::Section>& system_sections = ini.GetSections();
+
+    for (auto section : system_sections)
+    {
+      const std::string section_name = section.GetName();
+      if (section.HasLines())
+      {
+        // Trash INI File chunks
+        std::vector<std::string> chunk;
+        section.GetLines(&chunk, true);
+
+        if (chunk.size())
+        {
+          const auto mapped_config = MapINIToRealLocation(section_name, "");
+
+          if (mapped_config.section.empty() && mapped_config.key.empty())
+            continue;
+
+          auto* config_section =
+              config_layer->GetOrCreateSection(mapped_config.system, mapped_config.section);
+          config_section->SetLines(chunk);
+        }
+      }
+
+      // Regular key,value pairs
+      const IniFile::Section::SectionMap& section_map = section.GetValues();
+
+      for (auto& value : section_map)
+      {
+        const auto mapped_config = MapINIToRealLocation(section_name, value.first);
+
+        if (mapped_config.section.empty() && mapped_config.key.empty())
+          continue;
+
+        auto* config_section =
+            config_layer->GetOrCreateSection(mapped_config.system, mapped_config.section);
+        config_section->Set(mapped_config.key, value.second);
+      }
+    }
+
+    // Game INIs can have controller profiles embedded in to them
+    std::string num[] = {"1", "2", "3", "4"};
```

Comment:
`std::array`. This could also likely just be an array of individual chars instead of heap allocating four strings.